src/evaluator/data/vector_search.py

```
@@ -0,0 +1,101 @@
+
+        ranks = self.cross_encoder.rank(query, search_results)
+
+        final_results: List[str] = []
+        for rank in ranks[: self.max_results]:
+            element_idx: int | float | str = rank["corpus_id"]
+            try:
+                element_idx = int(element_idx)
+            except ValueError:
+                raise RuntimeError(
+                    f"'corpus_id' is not a valid integer. Using original value: {element_idx}"
+                )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)